dspace-server-webapp/src/test/java/org/dspace/app/rest/SubscriptionRestRepositoryIT.java

```diff
@@ -0,0 +1,847 @@
+        subscriptionParameterList.add(subscriptionParameter);
+        Subscription subscription = SubscribeBuilder.subscribeBuilder(context,
+                "TestType", publicItem, admin, subscriptionParameterList).build();
+        context.restoreAuthSystemState();
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String, Object> newSubscription = new HashMap<>();
+        newSubscription.put("subscriptionType", "test");
+        List<Map<String, Object>> list = new ArrayList<>();
+        Map<String, Object> sub_list = new HashMap<>();
+        sub_list.put("name", "frequency");
+        sub_list.put("value", "daily");
+        list.add(sub_list);
+        newSubscription.put("subscriptionParameterList", list);
+
+        getClient().perform(put("/api/core/subscriptions/" + subscription.getID())
+                   .param("dspace_object_id", publicItem.getID().toString())
+                   .param("eperson_id", admin.getID().toString())
+                   .content(objectMapper.writeValueAsString(newSubscription))
+                   .contentType(MediaType.APPLICATION_JSON_VALUE))
+                   .andExpect(status().isUnauthorized());
+    }
+
+    @Test
+    public void putSubscriptionForbiddenTest() throws Exception {
+        context.turnOffAuthorisationSystem();
+        List<SubscriptionParameter> subscriptionParameterList = new ArrayList<>();
+        SubscriptionParameter subscriptionParameter = new SubscriptionParameter();
+        subscriptionParameter.setName("Parameter1");
+        subscriptionParameter.setValue("ValueParameter1");
+        subscriptionParameterList.add(subscriptionParameter);
+        Subscription subscription = SubscribeBuilder.subscribeBuilder(context,
+                                    "TestType", publicItem, admin, subscriptionParameterList).build();
+        context.restoreAuthSystemState();
+
+        ObjectMapper objectMapper = new ObjectMapper();
+        Map<String, Object> newSubscription = new HashMap<>();
+        newSubscription.put("subscriptionType", "test");
+        List<Map<String, Object>> list = new ArrayList<>();
+        Map<String, Object> sub_list = new HashMap<>();
+        sub_list.put("name", "Frequency");
+        sub_list.put("value", "Daily");
+        list.add(sub_list);
+        newSubscription.put("subscriptionParameterList", list);
+
+        String token = getAuthToken(eperson.getEmail(), password);
+        getClient(token).perform(put("/api/core/subscriptions/" + subscription.getID())
+                        .param("eperson_id", admin.getID().toString())
+                        .param("dspace_object_id", publicItem.getID().toString())
+                        .content(objectMapper.writeValueAsString(newSubscription))
+                        .contentType(MediaType.APPLICATION_JSON_VALUE))
+                        .andExpect(status().isForbidden());
+    }
+
+    @Test
+    public void putSubscriptionTest() throws Exception {
```

Comment:
We should have a test that verifies a user cannot change the Subscription to any other Object or any other EPerson.  I think that's currently possible in this PR's code.

E.g.
* Create a subscription under one Eperson/Object
* Edit it and change it to a restricted Object (one you don't have access to) or an EPerson who you don't have access too.

Reply:
@tdonohue I have removed that business logic so not more: https://github.com/DSpace/DSpace/pull/8620/commits/00b53923a682391333415547f70874e7ad02634e#diff-7b083a954bbeb1f357856c1cb0872ab5bec3df518f0f64a5c7727ed1bc44fc2bL228-L229

Reply:
While you removed the logic, the tests all still pass in these params to the `PUT` command. We need to modify the tests to show the expected behavior... if it's no longer expected to pass these params to `PUT`, then they need to be removed from all tests.